sbi/neural_nets/flow_matcher.py

```diff
@@ -0,0 +1,230 @@
+
+    Args:
+        batch_x: Batch of xs, used to infer dimensionality and (optional) z-scoring.
+        batch_y: Batch of ys, used to infer dimensionality and (optional) z-scoring.
+        z_score_x: Whether to z-score xs passing into the network, can be one of:
+            - `none`, or None: do not z-score.
+            - `independent`: z-score each dimension independently.
+            - `structured`: treat dimensions as related, therefore compute mean and std
+            over the entire batch, instead of per-dimension. Should be used when each
+            sample is, for example, a time series or an image.
+        z_score_y: Whether to z-score ys passing into the network, same options as
+            z_score_x.
+        hidden_units: Number of hidden units in the ResNet blocks.
+        num_blocks: Number of ResNet blocks in the network.
+        num_freqs: Number of frequencies in the time embeddings.
+        embedding_net: Embedding network for batch_y.
+        kwargs: Additional keyword arguments passed to the FlowMatchingEstimator.
+    """
+    # Infer the output dimensionality of the embedding_net by making a forward pass.
+    check_data_device(batch_x, batch_y)
+    x_numel = get_numel(batch_x)  # theta
+    y_numel = get_numel(batch_y, embedding_net=embedding_net)  # x
+
+    # create the ResNet with GLU conditioning network
+    # Note, here we condition on theta and time (and not x) using GLUs
+    vectorfield_net = ResNetWithGLUConditioning(
+        input_dim=y_numel,  # the resnet takes the embedded x as input
+        hidden_units=hidden_features,
+        num_blocks=num_blocks,
+        # we condition on theta and embedded time using GLUs
+        condition_dim=(x_numel + 2 * num_freqs),
+        output_dim=x_numel,
+    )
+
+    # input data is only z-scored, not embedded.
+    z_score_x_bool, structured_x = z_score_parser(z_score_x)
+    if z_score_x_bool:
+        t_mean, t_std = z_standardization(batch_x, structured_x)
+        z_score_transform = torch.distributions.AffineTransform(
+            -t_mean / t_std, 1 / t_std
+        )
+    else:
+        z_score_transform = zuko.transforms.IdentityTransform()
+
+    # pre-pend the z-scoring layer to the embedding net.
+    z_score_y_bool, structured_y = z_score_parser(z_score_y)
+    if z_score_y_bool:
+        embedding_net = nn.Sequential(
+            standardizing_net(batch_y, structured_y), embedding_net
+        )
+
+    # create the flow matching estimator, will take care of time embeddings.
+    return FlowMatchingEstimator(
+        net=vectorfield_net,
+        input_shape=batch_x[0].shape,  # theta
+        condition_shape=batch_y[0].shape,  # x
+        zscore_transform_input=z_score_transform,
+        embedding_net=embedding_net,  # embedding net for x
+        num_freqs=num_freqs,  # number of frequencies in time embeddings
+        **kwargs,  # e.g., noise_scale
+    )
+
+
+class ResNetWithGLUConditioning(VectorFieldNet):
```

Comment:
I find it odd that this lives in `flow_matcher.py` but `ResNetBlock` lives in `embedding_nets.py`. Is there really a reason for this?